scripts/bootstrap_vm.sh

```diff
@@ -123,83 +123,94 @@ check_download_url () {
-elif [[ "$OS_TYPE" == *"Red Hat Enterprise"* && "$OS_TYPE" == *"8."* ]]; then
+  elif [[ "$OS_TYPE" == *"Red Hat Enterprise"* && "$OS_TYPE" == *"8."* ]]; then
     # Set for RHEL8 agent (RPM)
     FILE_DESCRIPTION="Red Hat ES 8, 9 / Alma Linux 8, 9 / Rocky Linux 8, 9 / Oracle Linux 8, 9 / (including Unbreakable Enterprise Kernel) (x86_64)"
     INSTALL_FILE="nessusagent.rpm"
     id="$(get_download_id "$FILE_DESCRIPTION")"
     DOWNLOAD_URL=$(check_download_url "$id")
-else
+  else
     # Set for Ubuntu agent (deb) AMD64
     FILE_DESCRIPTION="Ubuntu 14.04, 16.04, 18.04, 20.04, 22.04 (amd64)"
     INSTALL_FILE="nessusagent.deb"
     id="$(get_download_id "$FILE_DESCRIPTION")"
     DOWNLOAD_URL=$(check_download_url "$id")
-fi
+  fi
 
-# Install nessus agent
-curl --retry 3 -# -L -k -o $INSTALL_FILE $DOWNLOAD_URL
-if [[ "$OS_TYPE" == *"Red Hat Enterprise Linux"* ]]; then
-    /opt/nessus_agent/sbin/nessuscli agent status || rpm -Uh nessusagent.rpm
+  # Install nessus agent
+  curl --retry 3 -# -L -k -o $INSTALL_FILE $DOWNLOAD_URL
+
+  if [[ "$OS_TYPE" == *"Red Hat Enterprise Linux"* ]]; then
+    if [ ! -f "/opt/nessus_agent/sbin/nessuscli" ]; then
+      rpm -Uh nessusagent.rpm
+    fi
     rm -rf nessusagent.rpm
-else
-    /opt/nessus_agent/sbin/nessuscli agent status || dpkg -i nessusagent.deb
+  else
+    if [ ! -f "/opt/nessus_agent/sbin/nessuscli" ]; then
+      dpkg -i nessusagent.deb
+    fi
     rm -rf nessusagent.deb
-fi
+  fi
+
+  # Check if nessuscli exists
+  if [ ! -f "/opt/nessus_agent/sbin/nessuscli" ]; then
+    echo "Error: Nessus agent installation failed. /opt/nessus_agent/sbin/nessuscli not found."
+    exit 1
+  fi
 
-# Start Service
-/sbin/service nessusagent start
+  # Start Service
+  /sbin/service nessusagent start
 
-# Link agent
-NESSUS_STATUS=$(/opt/nessus_agent/sbin/nessuscli agent status -a | grep "Link status" | cut -f2 -d: | sed -e 's/^[[:space:]]*//')
-if [[ "$NESSUS_STATUS" == "Connected to"* ]]; then
+  # Link agent
+  NESSUS_STATUS=$(/opt/nessus_agent/sbin/nessuscli agent status -a | grep "Link status" | cut -f2 -d: | sed -e 's/^[[:space:]]*//')
+  if [[ "$NESSUS_STATUS" == "Connected to"* ]]; then
     echo $NESSUS_STATUS
-else
+  else
     echo "Connecting..."
     /opt/nessus_agent/sbin/nessuscli agent link --key=$KEY --groups=$GROUPS --host=$SERVER --port=8834
-fi
+  fi
 }
 
 # Exit on error
 set +e
 
-if [ "${UF_INSTALL}" = "true" ]
+if [ "true" = "true" ]
 then
-  install_splunk_uf "${UF_USERNAME}" "${UF_PASSWORD}" "${UF_PASS4SYMMKEY}" "${UF_GROUP}"
+  install_splunk_uf "hmcts_soc_admin" "BNFJN0XWmkCJw0S78HLO" "$7$oNroXETNQ9CvUjVvOv+gF4vnXZMUcOV5AYqzGX8Dl4QPUgp1xrUu68laynlhotZaNcOhCQ/FXXYtSH1VehZ5YA==" "hmcts_forwarders"
 fi
 
-if [ "${NESSUS_INSTALL}" = "true" ]
+if [ "true" = "true" ]
 then
-  install_nessus "${NESSUS_SERVER}" "${NESSUS_KEY}" "${NESSUS_GROUPS}"
+  install_nessus "nessus-scanners-prod000005.platform.hmcts.net" "86395175d0e6cf61a8915d84180ed64aa8e0e0ce38e52c64fd4bc8cb29418aea" "Prod-test"
```

Comment:
why is the key in plain text?

Reply:
reverted back to the original code to remove plain text